rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -699,13 +726,11 @@ macro_rules! hash_array {
 }
 
 /// Creates hash values for every element in the row based on the values in the columns
-pub fn create_hashes(
+pub fn create_hashes<'a>(
     arrays: &[ArrayRef],
     random_state: &RandomState,
-) -> Result<Vec<u64>> {
-    let rows = arrays[0].len();
-    let mut hashes = vec![0; rows];
-
+    hashes: &'a mut Vec<u64>,
```

Comment:
I think it would help readability to name this  `hashes_buffer ` for consistency